examples/simple_dff/test_dff.py

```diff
@@ -16,8 +16,9 @@
 async def dff_simple_test(dut):
     """Test that d propagates to q"""
 
-    # Assert initial output is unknown
-    assert LogicArray(dut.q.value) == LogicArray("X")
+    # Assert initial output is unknown, but not all simulators support 'X'
+    if not cocotb.SIM_NAME.lower().startswith("verilator"):
+        assert LogicArray(dut.q.value) == LogicArray("X")
```

Comment:
Verilator doesn't support unknown values:

https://verilator.org/guide/latest/languages.html#unknown-states

```
     0.00ns INFO     cocotb.regression                  dff_simple_test failed
                                                        Traceback (most recent call last):
                                                          File "/Users/dave/work/cocotb/examples/simple_dff/test_dff.py", line 21, in dff_simple_test
                                                            assert LogicArray(dut.q.value) == LogicArray("X")
                                                        AssertionError: assert LogicArray('0', Range(0, 'downto', 0)) == LogicArray('X', Range(0, 'downto', 0))
                                                         +  where LogicArray('0', Range(0, 'downto', 0)) = LogicArray(0)
                                                         +    where 0 = ModifiableObject(dff.q).value
                                                         +      where ModifiableObject(dff.q) = HierarchyObject(dff).q
                                                         +  and   LogicArray('X', Range(0, 'downto', 0)) = LogicArray('X')
```

Reply:
I think I'd prefer to just omit this check rather than have a simulator conditional in our "simple" example.